cmd/opentelemetry/app/exporter/elasticsearchexporter/esmodeltranslator/modeltranslator.go

```diff
@@ -134,7 +134,7 @@ func toTime(nano pdata.TimestampUnixNano) time.Time {
 func references(links pdata.SpanLinkSlice, parentSpanID pdata.SpanID, traceID dbmodel.TraceID) ([]dbmodel.Reference, error) {
 	parentSpanIDSet := len(parentSpanID.Bytes()) != 0
 	if !parentSpanIDSet && links.Len() == 0 {
-		return nil, nil
+		return []dbmodel.Reference{}, nil
```

Comment:
we could avoid some memory allocations by using constants in these cases, like
```
const emptyReferenceList = []dbmodel.Reference{}
```
the most impact will be for logs, since it's quite rare to find spans without references or tags.

BTW, I wonder if there is another bug in the translator from OTEL that fails to create a CHILD_OF reference to the parent span ID, which is required in the Jaeger model since we don't store parentSpanID in the database.

Reply:
Thanks for your feedback, i made changes accordingly.
I don't see where there could be a bug (...or i don't get it :smile: ). The `CHILD_OF` refs are tested properly as far as i can tell. It either creates a reference or it errors out.